### ISSUE TITLE
Extending profile module issues

### ISSUE THREAD

**Issue by VergilUa** [alvyxaz/barebones-masterserver#111]
Hi there, I'm trying to extend a profile module, and it seems there's no way to obtain a profile with Msf.Server.Profiles.FillProfileValues method, since there's no connection (because this is an extended server module). And it also seems there isn't a way to properly override methods to obtain ObservableServerProfile without breaking in the framework internals. Am I missing something?  

What I'm trying to do is to retrieve profile and modify some of the values send by a custom message. But I don't really know how to obtain the profile properly on the master server without breaking callbacks.

TL;DR: I wrote a method and put it into ProfileModule.cs:
```
 protected virtual ObservableServerProfile GetProfile(string username) {
            ObservableServerProfile profile;
            _profiles.TryGetValue(username, out profile);

            return profile;
        }
```

If I use this method only on server in (ExtendedProfileModule), will it get me into any troubles?

**Comment by alvyxaz** [comment]
Hey!

Nah, it shouldn't get you into trouble ;)

Actually, this method could be useful in the official release. If you want, make it **public** and submit a 
 commit (pull request) with it, and I'll approve it. If you're not sure how to do it, I can do it for you ;)

**Comment by VergilUa** [comment]
Ok, thanks for reply. It's been awhile since I've used github pull requests, so I'll pass. Feel free to add it though.

Thanks again :)

**Comment by lazalong** [comment]
If I am not mistaken this method wasn't yet added to the ProfileModule.
